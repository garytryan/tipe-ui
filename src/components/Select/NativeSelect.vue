<template>
  <select 
    :disabled="disabled" 
    :data-tipe-ui="$options.name" >
    <option 
      v-if="placeholder"
      :selected="true"
      disabled>{{ placeholder }}</option>
    <option 
      v-for="option in options"
      :value="option.value" 
      :selected="selectedValue.value === option.value"
      :key="option.value"
    >{{ option.label }}</option>
  </select>
</template>


<script>
import vueTypes from 'vue-types'
import SelectOptionShape from '@/types/SelectOptionShape'

export default {
  name: 'TipeNativeSelect',
  props: {
    disabled: vueTypes.bool,
    options: vueTypes.arrayOf(vueTypes.shape(SelectOptionShape)),
    placeholder: vueTypes.string,
    selectedValue: vueTypes.object
  }
}
</script>

<style lang="postcss">
select {
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}
</style>
