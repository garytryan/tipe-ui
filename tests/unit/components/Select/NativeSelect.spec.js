import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import NativeSelect from '@/components/Select/NativeSelect'

const props = {
  options: [
    { label: 'Owner', value: 'owner' },
    { label: 'Admin', value: 'admin' }
  ]
}
describe('NativeSelect.vue', () => {
  it('matches previous snapshot', async () => {
    const wrapper = shallowMount(NativeSelect)
    await expect(createRenderer().renderToString(wrapper.vm)).toMatchSnapshot()
  })
  it('has correct data-tipe-iu attibute', () => {
    const wrapper = shallowMount(NativeSelect)
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeNativeSelect')
  })
  it('should have the correct props', () => {
    const wrapper = shallowMount(NativeSelect, {
      propsData: {
        ...props,
        disabled: true,
        placeholder: 'Select a role',
        selectedValue: { label: 'Owner', value: 'owner' }
      }
    })
    expect(wrapper.props().options).toBe(props.options)
    expect(wrapper.props().disabled).toBe(true)
    expect(wrapper.props().placeholder).toBe('Select a role')
    expect(wrapper.props().selectedValue).toEqual({
      label: 'Owner',
      value: 'owner'
    })
  })
  it('should have the correct number of option tags', () => {
    const wrapper = shallowMount(NativeSelect, {
      propsData: {
        ...props
      }
    })
    const optionArr = wrapper.findAll('option')
    expect(optionArr).toHaveLength(2)
  })
  it('should have the correct number of option tags when a placeholder is present', () => {
    const wrapper = shallowMount(NativeSelect, {
      propsData: {
        ...props,
        placeholder: 'Select Something'
      }
    })
    const optionArr = wrapper.findAll('option')
    expect(optionArr).toHaveLength(3)
  })
})
