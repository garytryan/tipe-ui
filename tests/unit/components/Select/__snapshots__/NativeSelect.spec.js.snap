// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeSelect.vue matches previous snapshot 1`] = `Promise {}`;
